firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/analytics/AnalyticsConnectorAppExceptionEventRecorder.java

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics.internal.analytics;
+
+import android.os.Bundle;
+import com.google.firebase.analytics.connector.AnalyticsConnector;
+
+/** Records app exception events to the Firebase Analytics connector. */
+public class AnalyticsConnectorAppExceptionEventRecorder implements AppExceptionEventRecorder {
+
+  private static final String FIREBASE_CRASH_TYPE = "fatal";
+  private static final String FIREBASE_TIMESTAMP = "timestamp";
+  private static final String FIREBASE_APPLICATION_EXCEPTION = "_ae";
+  private static final String FIREBASE_ANALYTICS_ORIGIN_CRASHLYTICS = "clx";
```

Comment:
There's not a clear place to centralize this constant, so I kept it separate. Don't want to make it public on `FirebaseCrashlytics` or it would become part of the public API.

Reply:
I agree they shouldn't be part of the public API. Why not make them public in this (AnalyticsConnectorAppExceptionEventRecorder) class, or maybe in the parent interface? This implementation detail is essentially already exposed to FirebaseCrashlytics given that you need to know the value of this string within that class.

Reply:
I think of it a little differently. `FirebaseCrashlytics` needs this for *registration*. This class needs it for *sending*. The other place it's used it's for *listening*. So it wouldn't really make sense to expose it on the interface of any of those 3 behaviors for the others (specifically, in your suggestion, on the *sending* interface). You'd want it somewhere in a class that's one level up from all 3, and there's not a clear candidate for that.

Reply:
Hmmmm yeah I see your point. I begrudgingly accept your logic, sir.